odoo/addons/base/models/ir_translation.py

```diff
@@ -236,7 +236,7 @@ def _compute_source(self):
         '''
         for record in self:
             record.source = record.src
-            if record.type != 'model':
+            if record.type != 'model' or record.type == 'model' and not record.name:
```

Comment:
This is redundant. Your condition is `if A or (not A and B)`. As we are in the `or` clause, we know the first clause A was violated.